### PR TITLE
Exactness Criterion

### PR DESCRIPTION
**Pull request by gmourier**
Initialize exactness-criterion specification

### CODE REVIEW

**Comment by gmourier** [comment]
Thanks to @LegendreM the specification is already well advanced. The current [documentation page](https://docs.meilisearch.com/learn/core_concepts/relevancy.html#ranking-rules)  does not seem to need any update given its current state.

File: text/0036-exactness-criterion.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
+- Title: Exactness Criterion
+- Start Date: 2021-04-22
+- Specification PR: [#36](https://github.com/meilisearch/specifications/pull/36)
+- MeiliSearch Tracking-Issues:
```
Comment (curquiza) [inline text/0036-exactness-criterion.md:4]:
```suggestion
- MeiliSearch Tracking-Issues: [milli/#165](https://github.com/meilisearch/milli/issues/165)
```

File: text/0036-exactness-criterion.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+### VI. Impact on SDKs
+N/A
+
+## 2. Technical Aspects
+N/A
+
+## 3. Future Possibilities
+
+- In the second layer, we can rank `Any document that has an attribute that contains exactly the query` with the last word considered as prefix.
```
[the remark this answers is not in the document]
Comment (gmourier) [reply text/0036-exactness-criterion.md:147]: Thanks @LegendreM!

File: text/0036-exactness-criterion.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+has a rank of `1` (the user didn't finish to type the search but the final query could possibly be an exact match)
+
+##### Layer 3: Try to match exactly the word of the query anywhere in the document
+
+Any other document has a rank of `2 + 1` for each word that is not an exact match.
+
+A word is considered as an exact match when it matches:
+
+- without typo
+- without prefix
```
Comment (bidoubiwa) [inline text/0036-exactness-criterion.md:128]: What does `without prefix` mean?

**Review by bidoubiwa — APPROVED** [review]

File: text/0036-exactness-criterion.md (hunk 3 above)
Comment (LegendreM) [reply text/0036-exactness-criterion.md:128]: No word can match as a prefix. If my query contains `many` and a document contains `manythefish`, It's not counted as an exact match.

Comment (gmourier) [reply text/0036-exactness-criterion.md:128]: Thanks @LegendreM!

**Review by curquiza — APPROVED** [review]

**Review by ManyTheFish — APPROVED** [review]

File: text/0036-exactness-criterion.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+# Exactness Criterion
+
+## 1. Functional Specification
+
+### I. Summary
+
+Exactness Criterion is used within ranking rules to sort results by the similarity of the matched words with the query words. Documents that contain exactly the same terms as the ones queried first are placed first.
+
+> Ranking rules are built-in rules that ensure relevancy in search results. Ranking rules are applied in a default order which can be changed in the settings. You can add or remove rules and change their order of importance.
```
Comment (Kerollmops) [inline text/0036-exactness-criterion.md:14]:
```suggestion
> Ranking rules are built-in rules that ensure relevancy in search results. Ranking rules are applied in a default order which can be changed in the settings. You can add or remove rules and change their order.
```

File: text/0036-exactness-criterion.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+If documents have the exact same `_text_match` score, TypeSense accepts two more numerical fields in the `sort_by` parameter to break the tie.
+
+E.g. ```sort_by=_text_match:desc,price:desc,rating:desc```
+
+It is possible to put a numerical field before the `_text_match` score.
+
+E.g. ```sort_by=price:desc,_text_match:desc```
+
+```sort_by``` parameter is set at query time.
```
Comment (Kerollmops) [inline text/0036-exactness-criterion.md:38]:
```suggestion
E.g. `sort_by=_text_match:desc,price:desc,rating:desc`

It is possible to put a numerical field before the `_text_match` score.

E.g. `sort_by=price:desc,_text_match:desc`

`sort_by` parameter is set at query time.
```

**Review by Kerollmops — APPROVED** [review]
Thank you!

**Pull request merged**